kblocks/trainables/core.py

```diff
@@ -36,7 +36,7 @@ class Trainable:
     steps_per_epoch: Optional[int] = None
     validation_data: Optional[tf.data.Dataset] = None
     validation_steps: Optional[int] = None
-    callbacks: Tuple[tf.keras.callbacks.Callback, ...] = ()
+    callbacks: Iterable[tf.keras.callbacks.Callback] = ()
```

Comment:
I actually prefer this as a `Tuple` - but should update call sites. I know most attributes contained have their own mutable state, but I want to make `Trainable` as immutable as possible.

Reply:
Gotcha, well as long as it's made this way consistently…